File: September-LeetCoding-Challenge/09-Compare-Version-Numbers/Compare-Version-Numbers.cs
public class Solution {
    public int CompareVersion(string version1, string version2) {       
        var v1parts = version1.Split(".").ToList();
        var v2parts = version2.Split(".").ToList();
        
        if(v1parts.Count != v2parts.Count){
            while(v1parts.Count < v2parts.Count)
            {
                v1parts.Add("0");
            }
            while(v1parts.Count > v2parts.Count)
            {
                v2parts.Add("0");
            }
        }
        
        for(int i=0; i<v1parts.Count; i++){
            var v1 = int.Parse(v1parts[i]);
            var v2 = int.Parse(v2parts[i]);
            if(v1 > v2){
                return 1;
            }else if(v1 == v2){
                continue;
            }else if(v1 < v2){
                return -1;
            }
        }
        
        return 0;
    }
}

File: September-LeetCoding-Challenge/09-Compare-Version-Numbers/Compare-Version-Numbers.rb
def compare_version(version1, version2)
    arr1 = version1.split(".")
    arr2 = version2.split(".")
    
    len = [arr1.length-1, arr2.length-1].max

    0.upto(len) do |n|
      if arr1[n].nil? && arr2[n].nil?
        return 0
      else
        a = arr1[n] || "0"
        b = arr2[n] || "0"
        if a.to_i > b.to_i
          return 1
        elsif a.to_i < b.to_i
          return -1
        else
          next
        end
      end
    end

end